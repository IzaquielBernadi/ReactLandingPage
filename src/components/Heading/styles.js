import styled, { css } from 'styled-components';

const titleSizes = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lowerThenMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const isUppercase = (transform) => css`
  text-transform: ${transform ? 'uppercase' : 'none'};
`;

const setTextColor = (color, theme) => css`
  color: ${color ? theme.colors.light : theme.colors.primaryColor};
`;

export const Title = styled.h1`
  ${({ theme, fontColor, size, uppercase }) => css`
    ${setTextColor(fontColor, theme)};
    ${isUppercase(uppercase)};
    ${titleSizes[size](theme)};
  `}
`;
