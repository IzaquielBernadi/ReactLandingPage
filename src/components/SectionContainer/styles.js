import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme, background }) => css`
    max-width: 120rem;
    margin: 0 auto;
    padding: ${theme.space.sizes.large};
  `}
`;
