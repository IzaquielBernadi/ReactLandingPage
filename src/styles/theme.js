export const theme = {
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#dc143c',
    light: '#FFFFFF',
    mediumGray: '#DDDDDD',
  },
  font: {
    family: {
      openSans: "'Open Sans', sans-serif;",
      Montserrat: "'Montserrat', sans-serif;",
    },
    sizes: {
      xsmall: '8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
  },
  media: {
    lowerThanMedium: '(max-width: 768px)',
  },
  space: {
    sizes: {
      xsmall: '8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
  },
};
