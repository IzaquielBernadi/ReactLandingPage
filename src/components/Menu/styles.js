import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    border-bottom: ${theme.colors.mediumGray};
    background: ${theme.colors.light};

    ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    &${Heading} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.lowerThanMedium} {
      height: 100vh;

      ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: auto;
      }

      &${Heading} {
        padding-bottom: ${theme.space.sizes.large};
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media ${theme.media.lowerThanMedium} {
      display: block;
      text-align: center;
      padding: ${theme.space.sizes.xxlarge} 0;
    }
  `}
`;

export const Button = styled.div`
  ${({ theme }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.light};
    border: 1px solid red;
  `}

  > svg {
    width: 2.5rem;
    height: 2.5rem;
  }
`;
