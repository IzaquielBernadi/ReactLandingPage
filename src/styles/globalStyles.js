import styled, { createGlobalStyle, css } from 'styled-components';
import { theme } from './theme';

export const GlobalStyle = createGlobalStyle`

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html {
  font-size: 62.5%;
}

body {
  font-size: 1.6rem;
  font-family: ${theme.font.family.openSans};
}

h1, h2, h3, h4, h5, h6 {
  font-family: ${theme.font.family.Montserrat};
}

`;
