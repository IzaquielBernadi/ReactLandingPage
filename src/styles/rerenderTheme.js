import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const rerenderTheme = (children) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};
