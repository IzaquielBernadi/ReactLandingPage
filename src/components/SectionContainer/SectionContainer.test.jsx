import { render, screen } from '@testing-library/react';
import { SectionContainer } from '.';
import { renderTheme } from '../../styles/renderTheme';

describe('<SectionContainer />', () => {
  it('should render', () => {
    const { container } = renderTheme(
      <SectionContainer>
        <h1>Children</h1>
      </SectionContainer>,
    );
    expect(screen.getByRole('heading', { name: 'Children' })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
