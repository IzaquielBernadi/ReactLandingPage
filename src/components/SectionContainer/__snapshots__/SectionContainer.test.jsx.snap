// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionContainer /> should render 1`] = `
<div>
  <div
    class="sc-bcXHqe xjnZG"
  >
    <h1>
      Children
    </h1>
  </div>
</div>
`;
